packages/web-wallet/src/pages/Home.tsx

```diff
@@ -32,18 +32,26 @@ const Home: React.FC = () => {
     e.preventDefault();
     await requestSnap();
 
-    const viewKey = (await invokeSnap({ method: 'getViewingKey' })) as string;
-    console.log(viewKey);
+    const viewingKey = (await invokeSnap({ method: 'getViewingKey' })) as string;
+    const customBirthdayBlock = (await invokeSnap({ method: 'setBirthdayBlock' })) as number | null;
 
-    await addNewAccountFromUfvk(viewKey, birthdayHeight);
-    setBirthdayHeight(0);
+    const creationBlockHeight = typeof customBirthdayBlock === 'number' ? customBirthdayBlock : birthdayHeight
+
+    await addNewAccountFromUfvk(viewingKey, creationBlockHeight);
 
     await flushDbToStore();
   };
 
   useEffect(() => {
-    if (installedSnap) navigate('/dashboard/account-summary');
-  }, [installedSnap, navigate]);
+    if (installedSnap) {
+      const homeReload = async () => {
+        const accountData = await getAccountData();
```

Comment:
Shouldn't we get Account data only on the Receive page?

Reply:
I replaced unifiedAddress instead installedSnap as a flag that allows `/dashboard/account-summary`.

Maybe checking wallet summary is better for this case